File: apps/visitor/types/scene.ts
export type SceneFloor = {
  piso_id: number;
  codigo: string;
  origin: [number, number, number];
  axis_x: [number, number, number];
  axis_y: [number, number, number];
};
export type Scene = {
  shopping_id: number;
  scene_version: string;
  release: number;
  model_url: string;
  floors: SceneFloor[];
  pois: { loja_id: number; codigo: string; object_prefix: string; anchor_node_id: number }[];
  anchors: { node_id: number; codigo: string; piso_id: number; coord_x: number; coord_y: number; type: string; name: string }[];
};

File: services/api/app/schemas/scene.py
from pydantic import BaseModel, Field


class SceneFloor(BaseModel):
    piso_id: int
    codigo: str
    origin: list[float] = Field(min_length=3, max_length=3)
    axis_x: list[float] = Field(min_length=3, max_length=3)
    axis_y: list[float] = Field(min_length=3, max_length=3)


class ScenePoi(BaseModel):
    loja_id: int
    codigo: str
    object_prefix: str
    anchor_node_id: int


class SceneAnchor(BaseModel):
    node_id: int
    codigo: str
    piso_id: int
    coord_x: float = Field(ge=0, le=1)
    coord_y: float = Field(ge=0, le=1)
    type: str
    name: str


class SceneResponse(BaseModel):
    shopping_id: int
    scene_version: str
    release: int
    model_url: str
    model_sha256: str
    floors: list[SceneFloor]
    pois: list[ScenePoi]
    anchors: list[SceneAnchor]
